### PR TITLE
Add stats pulling to Gitlic

### PR DESCRIPTION
**Pull request by maxgardner**
I imagine there definitely are ways to optimize this.

### CODE REVIEW

File: swgithub/main.go [hunk 1]
```diff
@@ -22,87 +24,73 @@ func getIncludedMap(a []string) map[string]bool {
...
-		if includedOrgs == nil {
-			validOrgs = orgs
-			break
-		}
-		if _, ok := includedOrgs[strings.ToLower(*org.Login)]; ok {
-			validOrgs = append(validOrgs, org)
-		} else {
-			log.Printf("Ignored %s\n", *org.Login)
+func saveRepoStats(ctx context.Context, fo map[string]*os.File, org string, repo string, commits []*github.WeeklyCommitActivity, mods []*github.WeeklyStats) {
+	totalCommits := 0
```
Comment (seubert) [inline swgithub/main.go]: You can collapse all these to one line with `var totalCommits, linesAdded (...) int`

File: swgithub/main.go [hunk 2]
```diff
@@ -22,87 +24,73 @@ func getIncludedMap(a []string) map[string]bool {
...
+	mostCommitsOneDay := 0
+	mostCommitsOneWeek := 0
+	var mostCommittable string
+	highestDayCounts := map[string]int{"Sunday": 0, "Monday": 0, "Tuesday": 0, "Wednesday": 0, "Thursday": 0, "Friday": 0, "Saturday": 0}
+
+	for _, commit := range commits {
+		totalCommits += *commit.Total
+
+		if *commit.Total > 0 {
+			highestDay := 0
```
Comment (seubert) [inline swgithub/main.go]: same as above

File: swgithub/github.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/solarwinds/gitlic-check/config"
+)
+
+func GetSWOrgs(ctx context.Context, ghClient *github.Client, cf config.Config) ([]*github.Organization, error) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetSWOrgs should have comment or be unexported (golint)

File: swgithub/github.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+		if _, ok := includedOrgs[strings.ToLower(*org.Login)]; ok {
+			validOrgs = append(validOrgs, org)
+		} else {
+			log.Printf("Ignored %s\n", *org.Login)
+		}
+	}
+	return validOrgs, nil
+}
+
+func GetOrgInvites(ctx context.Context, ghClient *github.Client, org *github.Organization) ([]*github.Invitation, error) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetOrgInvites should have comment or be unexported (golint)

File: swgithub/github.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+		if resp.NextPage == 0 {
+			opt.Page = 1
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+	return invites, nil
+}
+
+func GetOrgMembers(ctx context.Context, ghClient *github.Client, org *github.Organization, opt *github.ListMembersOptions) ([]*github.User, error) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetOrgMembers should have comment or be unexported (golint)

File: swgithub/github.go [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+		if resp.NextPage == 0 {
+			opt.Page = 1
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+	return members, nil
+}
+
+func GetOrgOwners(ctx context.Context, ghClient *github.Client, org *github.Organization) ([]*github.User, error) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetOrgOwners should have comment or be unexported (golint)

File: swgithub/github.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+		if resp.NextPage == 0 {
+			memOpt.Page = 1
+			break
+		}
+		memOpt.Page = resp.NextPage
+	}
+	return members, nil
+}
+
+func GetOrgRepositories(ctx context.Context, ghClient *github.Client, org *github.Organization) ([]*github.Repository, error) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetOrgRepositories should have comment or be unexported (golint)

File: swgithub/github.go [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+			repoOpt.Page = 1
+			break
+		}
+		repoOpt.Page = resp.NextPage
+	}
+
+	return repos, nil
+}
+
+func GetCommitActivity(ctx context.Context, ghClient *github.Client, org string, repo string) ([]*github.WeeklyCommitActivity, error, bool) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: error should be the last type when returning multiple items (golint)

Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetCommitActivity should have comment or be unexported (golint)

File: swgithub/github.go [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+	act, resp, err := ghClient.Repositories.ListCommitActivity(ctx, org, repo)
+	if resp.StatusCode == 202 {
+		return nil, nil, true
+	} else if err != nil && resp.StatusCode != 202 {
+		return nil, err, true
+	}
+	return act, nil, false
+}
+
+func GetAdditionsDeletions(ctx context.Context, ghClient *github.Client, org string, repo string) ([]*github.WeeklyStats, error, bool) {
```
Comment (Beaujr) [inline swgithub/github.go]: warning: exported function GetAdditionsDeletions should have comment or be unexported (golint)

**Comment by trevrosen** [comment]
If we're going to add these lint stats, they should really be from a bot user and not just always Beau 😂

**Comment by maxgardner** [comment]
Working on this from new branch feat/ts-152/stats-report.

**Pull request closed without merging**